plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt

```diff
@@ -110,6 +115,18 @@ abstract class LoginBrowser(
         else -> ""
     }
 
+    private fun isReAuth(scopes: List<String>): Boolean = if (scopes.toSet().intersect(Q_SCOPES.toSet()).isNotEmpty()) {
+        ToolkitConnectionManager.getInstance(project).activeConnectionForFeature(QConnection.getInstance()) != null
```

Comment:
nit: can we initialize all the connections before so we can reuse the variables?

Reply:
Pulled `ToolkitConnectionManager.getInstance(project)` to the top. Connections are branch dependent so initializing at the top is undesirable.